internal/scheduler/submitcheck_test.go

```diff
@@ -0,0 +1,175 @@
+			expectPass:     false,
+		},
+		"no jobs schedule due to resources": {
+			executorTimout: defaultTimeout,
+			config:         testSchedulingConfig(),
+			executors:      []*schedulerobjects.Executor{testExecutor(baseTime)},
+			jobs:           []*api.Job{test100CoreCpuJob()},
+			expectPass:     false,
+		},
+		"no jobs schedule due to selector": {
+			executorTimout: defaultTimeout,
+			config:         testSchedulingConfig(),
+			executors:      []*schedulerobjects.Executor{testExecutor(baseTime)},
+			jobs:           []*api.Job{test1CoreCpuJobWithNodeSelector(map[string]string{"foo": "bar"})},
+			expectPass:     false,
+		},
+		"no jobs schedule due to executor timeout": {
+			executorTimout: defaultTimeout,
+			config:         testSchedulingConfig(),
+			executors:      []*schedulerobjects.Executor{testExecutor(expiredTime)},
+			jobs:           []*api.Job{test1CoreCpuJob()},
+			expectPass:     false,
+		},
+		"multiple executors, 1 expired": {
+			executorTimout: defaultTimeout,
+			config:         testSchedulingConfig(),
+			executors:      []*schedulerobjects.Executor{testExecutor(expiredTime), testExecutor(baseTime)},
+			jobs:           []*api.Job{test1CoreCpuJob()},
+			expectPass:     true,
+		},
+		"gang job all jobs fit": {
+			executorTimout: defaultTimeout,
+			config:         testSchedulingConfig(),
+			executors:      []*schedulerobjects.Executor{testExecutor(baseTime)},
+			jobs:           testNJobGang(5),
+			expectPass:     true,
+		},
+		"gang job all jobs don't fit": {
+			executorTimout: defaultTimeout,
+			config:         testSchedulingConfig(),
+			executors:      []*schedulerobjects.Executor{testExecutor(baseTime)},
+			jobs:           testNJobGang(100),
+			expectPass:     false,
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+
+			ctrl := gomock.NewController(t)
+			mockExecutorRepo := schedulermocks.NewMockExecutorRepository(ctrl)
+			mockExecutorRepo.EXPECT().GetExecutors(ctx).Return(tc.executors, nil).AnyTimes()
+			fakeClock := clock.NewFakeClock(baseTime)
+			submitCheck := NewSubmitChecker(tc.executorTimout, tc.config, mockExecutorRepo)
+			submitCheck.clock = fakeClock
+			submitCheck.updateExecutors(ctx)
+			result, msg := submitCheck.CheckApiJobs(tc.jobs)
+			assert.Equal(t, tc.expectPass, result)
+			if !tc.expectPass {
+				assert.NotEqual(t, "", msg)
+			}
+			log.Info(msg)
+		})
+	}
+}
+
+func test1CoreCpuJob() *api.Job {
+	return &api.Job{
+		Id: util.NewULID(),
+		PodSpec: &v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Resources: v1.ResourceRequirements{
+						Limits: map[v1.ResourceName]resource.Quantity{
+							"cpu": resource.MustParse("1"),
+						},
+						Requests: map[v1.ResourceName]resource.Quantity{
+							"cpu": resource.MustParse("1"),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func testNJobGang(n int) []*api.Job {
```

Comment:
Move to `testfixtures_test.go` please.

Reply:
Even better, write tests to use existing test fixtures instead of writing new ones.

Reply:
These do the use the test fixtures- `test1CoreCpuJob()`, `testNodeItems1()` , `testGangIdAnnotation` etc, they just add extra information as needed for these tests.  I guess we could add helper methods to testFixtures to cover these cases, but I'm not convinced they're sufficiently generic yet.